service/history/task/task.go

```diff
@@ -271,6 +272,13 @@ func (t *taskImpl) HandleErr(err error) (retErr error) {
 		return err
 	}
 
+	// If the task list is not owned by the host we connected to we just return an error, so we retry in a bit
+	// with the new membership information.
+	var taskListNotOwnedByHostError *cadence_errors.TaskListNotOwnedByHostError
+	if errors.As(err, &taskListNotOwnedByHostError) {
+		return err
```

Comment:
Let's emit a count metric here similar to other error handling branches below. It would be useful to see trends of such cases (at the domain/tasklist level if possible)